### PR TITLE
Show overall household status, not HoH's status

### PR DESCRIPTION
**Pull request by martha**
## Description

GH issue: https://github.com/open-path/Green-River/issues/6337

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: graphql.schema.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
         "kind": "ENUM",
         "name": "ActiveStatus",
         "description": null,
+        "isOneOf": null,
```
Comment (gigxz) [inline graphql.schema.json]: opened https://github.com/greenriver/hmis-frontend/pull/872 for this unrelated change

File: src/modules/hmis/components/HouseholdStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { memo } from 'react';
+import EnrollmentStatus from '@/modules/hmis/components/EnrollmentStatus';
+import { HouseholdWithEnrollmentFieldsFragment } from '@/types/gqlTypes';
+
+interface Props {
+  household: HouseholdWithEnrollmentFieldsFragment;
+}
+
+const HouseholdStatus: React.FC<Props> = ({ household }) => {
+  // When showing the status for the household, show the 'most important' status - Incomplete, Open, or Exited.
```
Comment (gigxz) [inline src/modules/hmis/components/HouseholdStatus.tsx]: This duplicates logic that we already have in the [household view](https://github.com/greenriver/hmis-warehouse/blob/f4a3f8ca1712bd50f53506c178ebb68a2ec59444/db/views/hmis_households_v07.sql#L19). Sorry I did not mention that we already have it! 

I think it would be best to re-use that logic by having the household type resolve an `in_progress` bool (or `any_in_progress`?) that is resolved as `object.any_wip?`. It wouldn't hurt to resolve the other existing fields as well, in case we need them later:
```
household {
   anyInProgress // any_wip
   earliestEntryDate // earliest_entry
   latestExitDate // latest_exit
}
```

File: graphql.schema.json (hunk 1 above)
Comment (martha) [reply graphql.schema.json]: Thanks! 👍

File: src/modules/hmis/components/EnrollmentStatus.tsx [hunk 3]
```diff
@@ -2,75 +2,88 @@ import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
...
-        {!hideIcon && <Icon fontSize='small' />}
-        {text}
+        <statusProps.Icon fontSize='small' />
+        {statusProps.text}
       </Stack>
     </Typography>
   );
 };
 
+const EnrollmentStatus = ({
```
Comment (martha) [inline src/modules/hmis/components/EnrollmentStatus.tsx:57]: @gigxz this refactor removes some customizability that we weren't using, as far as I could tell:
- hideIcon
- withActiveRange
- activeColor
- closedColor

all were unused props. I can add back support for them if you prefer, let me know.

File: src/modules/hmis/components/HouseholdStatus.tsx (hunk 2 above)
Comment (martha) [reply src/modules/hmis/components/HouseholdStatus.tsx]: Good plan, thanks for this suggestion!!

File: src/modules/hmis/components/EnrollmentStatus.tsx (hunk 3 above)
Comment (gigxz) [reply src/modules/hmis/components/EnrollmentStatus.tsx:57]: great, thanks! no need to keep them around

File: src/modules/hmis/components/EnrollmentStatus.tsx [hunk 4]
```diff
@@ -2,75 +2,88 @@ import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
...
+
+export const HouseholdStatus = ({
+  household,
+}: {
+  household: HouseholdWithEnrollmentFieldsFragment;
+}) => {
+  if (household.anyInProgress) return <CommonStatus variant='inProgress' />;
+  if (!household.latestExitDate) return <CommonStatus variant='open' />;
+  return <CommonStatus variant='exited' />;
+};
```
Comment (gigxz) [inline src/modules/hmis/components/EnrollmentStatus.tsx:76]: Nice improvement

File: src/api/operations/staffAssignment.fragments.graphql [hunk 5]
```diff
@@ -40,11 +40,9 @@ fragment StaffAssignmentWithClients on StaffAssignment {
   assignedAt
   unassignedAt
   household {
-    id
+    ...HouseholdWithEnrollmentFields
```
Comment (gigxz) [inline src/api/operations/staffAssignment.fragments.graphql]: This looks like it resolves ssn and dob (and access object) for all household members which we don't need for staff assignment. I think we should avoid that since it's not necessary in this case and it is highly sensitive.

Maybe just inline the fields needed for staff assignment?

File: src/modules/hmis/components/EnrollmentStatus.tsx [hunk 6]
```diff
@@ -2,75 +2,88 @@ import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
...
+  if (enrollment.inProgress) return <CommonStatus variant='inProgress' />;
+  if (enrollment.autoExited) return <CommonStatus variant='autoExited' />;
+  if (!enrollment.exitDate) return <CommonStatus variant='open' />;
+  return <CommonStatus variant='exited' />;
+};
+
+export const HouseholdStatus = ({
+  household,
+}: {
+  household: HouseholdWithEnrollmentFieldsFragment;
```
Comment (gigxz) [inline src/modules/hmis/components/EnrollmentStatus.tsx]: Might be better to loosen these constraints a bit, since only 2 fields are needed. Then we can use this component in other places and not worry about type issues.

```suggestion
  household: Pick<Household, 'anyInProgress' | 'latestExitDate'>;
```
The same could be done for `EnrollmentStatus` component above as well. I think that approach would be more maintainable than jamming in all the fragments we happen to use. (I know, we do that in a lot of places)

File: graphql.schema.json [hunk 7]
```diff
@@ -12,7 +12,6 @@
         "kind": "ENUM",
         "name": "ActiveStatus",
         "description": null,
-        "isOneOf": null,
```
Comment (gigxz) [inline graphql.schema.json]: Can you try running `yarn` and then codegen again? I think these should be present.

**Review by gigxz — APPROVED** [review]
LGTM after fixing codegen changes!
